File: backend/app/schemas/market.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class MarketIndexOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    market: str
    index_value: float
    change: float
    change_rate: float
    foreign_net_buy: float | None
    institution_net_buy: float | None
    individual_net_buy: float | None
    total_trading_value: float
    data_source: str
    updated_at: datetime


class MarketOverviewOut(BaseModel):
    # 국내는 [KOSPI, KOSDAQ], 미국은 [NYSE, NASDAQ] 순서로 2개가 들어온다 - 고정된
    # kospi/kosdaq 필드 대신 리스트로 일반화해 국가가 늘어나도 스키마가 그대로다.
    indices: list[MarketIndexOut]
    country: str = "KR"
    foreign_net_buy_total: float | None
    institution_net_buy_total: float | None
    individual_net_buy_total: float | None
    total_trading_value: float
    updated_at: datetime
    data_source: str = "mock"

File: frontend/lib/market.ts
export type Market = "KOSPI" | "KOSDAQ" | "NYSE" | "NASDAQ";
export type Country = "KR" | "US";

export const MARKETS_BY_COUNTRY: Record<Country, [Market, Market]> = {
  KR: ["KOSPI", "KOSDAQ"],
  US: ["NYSE", "NASDAQ"],
};

const COUNTRY_BY_MARKET: Record<Market, Country> = {
  KOSPI: "KR",
  KOSDAQ: "KR",
  NYSE: "US",
  NASDAQ: "US",
};

const CURRENCY_BY_COUNTRY: Record<Country, "KRW" | "USD"> = { KR: "KRW", US: "USD" };

export function countryForMarket(market: Market): Country {
  return COUNTRY_BY_MARKET[market];
}

export function currencyForMarket(market: Market): "KRW" | "USD" {
  return CURRENCY_BY_COUNTRY[countryForMarket(market)];
}
